### ISSUE TITLE
Question: What do the numbers mean in the SQuAD return value?

### ISSUE THREAD

**Issue by jgostylo** [deepmipt/DeepPavlov#938]
I am just using the Docker image for SQuAD-en.  I have really long texts so I am chunking them up and asking the question to each chunk.  I assume the numbers that come back with the results are meaningful for ranking the returned answers but I have not found anything in the documentation telling me what those numbers represent.  I would not mind a bit of hand holding if there is a already an algorithm for using those number to rank the results.

Also, I want to say that the docker image is set to use BERT.  Is that correct?  I also assume that it is not configured for "noan".

**Comment by yurakuratov** [comment]
Hi!

In SQuAD-en [Dockerfile](https://github.com/deepmipt/stand_kubernetes_cluster/blob/master/models/stand_squad_en/Dockerfile) you can see that `squad/squad_bert_infer.json` configuration file is used. 

The model outputs answer, answer start position in characters and answer logits.

Yes, this model uses pre-trained BERT model.

As BERT model can work only with texts with length less than 512 subtokens, the `squad_bert_infer` model chunks input texts and outputs an answer with the highest answer score (multiplication of logits for answer start and end positions). If you want to select the best answer in different way, you should use `squad/squad_bert.json` model and implement selection method.

This model might output `no answer`, but is was not specially trained for it.

**Comment by jgostylo** [comment]
Thank you for the response!  I did not realize that the second return value (first number) was the character position.  I thought it was a different type of score or rating.  That makes sense though.  Also good to know that I can send text greater than 512 tokens and it will do the chunking for me.  I was attempting to break it up myself knowing that the bert model would only handle 512 token chunks.  Will it do this breaking it up by sentence?  If it just takes exactly 512 token chunks and then does the next I may want to keep mine where I make sure that sentences remain intact.

**Comment by yurakuratov** [comment]
Sentence boundaries are preserved by using `sent_tokenize` from `nltk` for english language. You can check it here: [deeppavlov/models/bert/bert_squad.py:266](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/models/bert/bert_squad.py#L266)
